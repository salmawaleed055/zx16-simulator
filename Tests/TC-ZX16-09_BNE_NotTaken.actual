Loaded 12 bytes into memory
0x0000: 1471    xori x1, 10
0x0002: 14B1    xori x2, 10
0x0004: 24C2    beq x3, x2, 0x0008
0x0006: 0271    xori x1, 1
0x0008: 0000    add x0, x0
0x000A: FFC7    Unknown opcode

--- Final State ---
x0: 0x0000
x1: 0x000b
x2: 0x000a
x3: 0x0000
x4: 0x0000
x5: 0x0000
a0: 0x0000
a1: 0x0000
PC: 0x000a
---------------------

Simulation finished.
