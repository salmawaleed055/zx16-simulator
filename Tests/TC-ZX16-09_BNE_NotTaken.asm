# TC-ZX16-09_BNE_NotTaken.s
# Test Objective: Validate BNE (Branch Not Equal) instruction - branch not taken
# x1=10, x2=10, branch to PC+2 (fall through)
li x1, 10
li x2, 10
bne x1, x2, +4 # Current PC + 4 = 0x0004 + 4 = 0x0008
li x1, 1       # Executed, PC becomes 0x0006
nop            # This instruction is now at PC 0x0008
ecall 0x3FF