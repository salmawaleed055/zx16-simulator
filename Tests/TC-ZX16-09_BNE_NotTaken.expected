0x0000: 1471    li x1, 10
0x0002: 14B1    li x2, 10
0x0004: 24C2    bne x1, x2, 0x0008
0x0006: 0271    li x1, 1
0x0008: 0000    add x0, x0
0x000A: FFC7    ecall 0x3FF

--- Final State ---
x0: 0x0000
x1: 0x0001
x2: 0x000A
x3: 0x0000
x4: 0x0000
x5: 0x0000
a0: 0x0000
a1: 0x0000
PC: 0x000C
---------------------
Simulation finished.